### DISCUSSION TITLE
Some application have no SSD

### DISCUSSION THREAD

**Discussion by ffs-akl** [labwc/labwc#2147] [Q&A] [answered]
Although the window rules have been defined in rc.xml, some programs still don't have server-side decorations.
For example, 
  ```
<windowRules>
     <windowRule title="lxqt-runner" serverDecoration="yes"/>
     <windowRule title="lxqt-runner" skipTaskbar="no" />
    </windowRules>

```

**Comment by stefonarch** [comment, marked as answer]
You can't use window rules for layer surfaces as `lxqt-runner` is one. The only compositor which has window rules for layers is hyprland afaik, which apply not to their name but scope (launcher, desktop, dock ecc)
`layerrule = noanim, launcher`

**Reply by ffs-akl** [reply]
Thank you very much!

**Reply by ffs-akl** [reply]
is there an app_id for lxqt-runner?  I didn't find the app_id by `wlrctl toplevel list  `

**Reply by stefonarch** [reply]
No, app_id is for windows, not for layers. It had one once in LXQt 1.* before we could use `layer-shell-qt`.  Hyprland can list them, runner, desktop, notifications and the desktop:

```
$ sleep 3 && notify-send test && sleep 1 && hyprctl layers
Monitor eDP-1:
        Layer level 0 (background):
                Layer 6070c82dc480: xywh: 0 1024 1920 1080, namespace: desktop
        Layer level 1 (bottom):
        Layer level 2 (top):
                Layer 6070c78fff10: xywh: 0 1024 1920 48, namespace: dock
                Layer 6070c78f4230: xywh: 1877 1194 43 788, namespace: dock
                Layer 6070c686e340: xywh: 161 2061 1555 43, namespace: dock
                Layer 6070c82380e0: xywh: 738.5 1291 400 46, namespace: launcher
        Layer level 3 (overlay):
                Layer 6070c6484840: xywh: 1341 1074 534 89, namespace: notification

```

**Reply by ffs-akl** [reply]
Thank your very much!
